scipy/stats/stats.py

```diff
@@ -3563,7 +3572,12 @@ def kendalltau(x, y, initial_lexsort=None, nan_policy='propagate', method='auto'
     elif contains_nan and nan_policy == 'omit':
         x = ma.masked_invalid(x)
         y = ma.masked_invalid(y)
-        return mstats_basic.kendalltau(x, y, method=method)
+        if variant == 'taua':            
+            return mstats_basic.kendalltau(x, y, method=method, use_ties=False)
+        elif variant == 'taub':
+            return mstats_basic.kendalltau(x, y, method=method, use_ties=True)
+        elif variant == 'tauc':
```

Comment:
Else? I.e. what if variant == 'xyz'? Suggest to mention that nans are in the data in the error message

Reply:
`else` was added. 
Re: nans, that is handled by `_contains_nan`